tensorflow_datasets/core/features/audio_feature.py

```diff
@@ -107,6 +107,19 @@ def repr_html(self, ex: np.ndarray) -> str:
         ' controlsList="nodownload" />'
     )
 
+  @classmethod
+  def from_json(cls, value) -> 'FeatureConnector':
+    pass
+
+  def to_json(self):
+    return {
+        'type': type(self).__name__,
```

Comment:
The `type` field seems to be duplicated across all features. As this is how feature class are detected, I think it is important to automate this field.

I think we should have two method `from_json` and `from_json_content`. Here is my proposal:

The `FeatureConnector` implement the 4 methods:

```py
def from_json(cls, value) -> 'FeatureConnector':
  subclass = cls._REGISTERED_FEATURES.get(value['type'])
  return subclass.from_json_content(value['content'])

def to_json(self, value):
  return {
      'type': type(self).__name__,
      'content': self.to_json_content(),
  }
  
  

def from_json_content(cls, value):
  if value:
    raise ValueError(
        f'FeatureConnector {cls} should overwrite `from_json_content`'
    )
  return cls()

def to_json_content(cls, value):
  return {}
```

Then subclasses would only implement the `to_json_content` and `from_json_content` optionally.

```py
def to_json(self):
  return {
        'file_format': self._file_format,
        'shape': list(self._shape),
        'dtype': self._dtype.name,
        'sample_rate': self._sample_rate,
    }
```

Reply:
Maybe the `from_json_content` in the base class could be simplified even further:

```py
# In features.py
def from_json_content(cls, value):
  return cls(**value)
```

This way, many FeatureConnector wouldn't have to implement `from_json_content` at all.